Code/GraphMol/new_canon.cpp

```diff
@@ -678,24 +679,51 @@ void updateAtomNeighborIndex(canon_atom *atoms, std::vector<bondholder> &nbrs) {
   std::sort(nbrs.begin(), nbrs.end(), bondholder::greater);
 }
 
+// This routine calculates the number of swaps that would be required to
+// determine what the smiles chirality value would be for a given chiral atom
+// given that the atom is visited first from the atom of interest.
+// THis is used to determine which of two atoms has priority based on the
+// neighbor's chirality
+//
+// If the chiral neighbor has two equivlent (at least so far) neighbors that are
+// not the atom of interest, it cannot be used to determine the priority of the
+// atom of interest.  For this reason, we keep track of the number of neighbors
+// that have the same priority so far.  If any two are the same, we do NOT use
+// that neighbor to determine the priority of the atom of interest.
+
 void updateAtomNeighborNumSwaps(
     canon_atom *atoms, std::vector<bondholder> &nbrs, unsigned int atomIdx,
     std::vector<std::pair<unsigned int, unsigned int>> &result) {
   bool isRingAtom = queryIsAtomInRing(atoms[atomIdx].atom);
   for (auto &nbr : nbrs) {
     unsigned nbrIdx = nbr.nbrIdx;
 
+    std::list<unsigned int> neighborsSeen;
+    bool tooManySimilarNbrs = false;
     if (isRingAtom && atoms[nbrIdx].atom->getChiralTag() != 0) {
       std::vector<int> ref, probe;
       for (unsigned i = 0; i < atoms[nbrIdx].degree; ++i) {
         ref.push_back(atoms[nbrIdx].nbrIds[i]);
```

Comment:
Might be more readable with

```
auto nbrId = atoms[atoms[nbrIdx].nbrIds[I]];
```

since this is used a lot in the code.
```

Reply:
I do not understand the comment.  And you might be referring to code I did not change.  Please advise.

Reply:
Lines 707, 709, 713 use the construct

```
atoms[nbrIdx].nbrIds[I]
```

this night be more readable as:


```
       const auto nbrid =  atoms[nbrIdx].nbrIds[I];
       if (atomIdx != nbrid {
           if ((std::find(neighborsSeen.begin(), neighborsSeen.end(),
                          atoms[nbrid.index) !=
                neighborsSeen.end())) {
             tooManySimilarNbrs = true;
           } else {
             neighborsSeen.push_back(atoms[nbrid].index);
           }
         }
```

Reply:
Got it - thanks,  I called the new variable nbrNbrId - it is the id of the neighbor's neighbors